runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -10393,227 +10393,637 @@ static TR::Register *inlineIsAssignableFrom(TR::Node *node, TR::CodeGenerator *c
-    dependencies->getPostConditions()->getRegisterDependency(7)->setExcludeGPR0(); // multiplierAddrReg
-
-    dependencies->addPostCondition(multiplierReg, TR::RealRegister::NoReg);
-    dependencies->addPostCondition(high4Reg, TR::RealRegister::NoReg);
-    dependencies->addPostCondition(low4Reg, TR::RealRegister::NoReg);
-    dependencies->addPostCondition(vtmp1Reg, TR::RealRegister::NoReg);
-    dependencies->addPostCondition(vtmp2Reg, TR::RealRegister::NoReg);
-    dependencies->addPostCondition(vconstant0Reg, TR::RealRegister::NoReg);
-    dependencies->addPostCondition(vconstantNegReg, TR::RealRegister::NoReg);
-    dependencies->addPostCondition(vunpackMaskReg, TR::RealRegister::NoReg);
-
-    generateDepLabelInstruction(cg, TR::InstOpCode::label, node, endLabel, dependencies);
-
-    node->setRegister(hashReg);
-    cg->decReferenceCount(valueNode);
-    cg->decReferenceCount(offsetNode);
-    cg->decReferenceCount(countNode);
-
-    cg->stopUsingRegister(valueReg);
-    cg->stopUsingRegister(endReg);
-    cg->stopUsingRegister(vendReg);
-    cg->stopUsingRegister(tempReg);
-    cg->stopUsingRegister(constant0Reg);
-    cg->stopUsingRegister(condReg);
-    cg->stopUsingRegister(multiplierAddrReg);
-
-    cg->stopUsingRegister(multiplierReg);
-    cg->stopUsingRegister(high4Reg);
-    cg->stopUsingRegister(low4Reg);
-    cg->stopUsingRegister(vtmp1Reg);
-    cg->stopUsingRegister(vtmp2Reg);
-    cg->stopUsingRegister(vconstant0Reg);
-    cg->stopUsingRegister(vconstantNegReg);
-    cg->stopUsingRegister(vunpackMaskReg);
-    return hashReg;
-}
+static TR::Register*
+hashCodeHelper(TR::Node *node, TR::CodeGenerator *cg, TR::DataType elementType,
+               TR::Node *hashNode, bool isSigned)
+   {
+   TR::Compilation *comp = cg->comp();
+   TR_J9VMBase *fej9 = (TR_J9VMBase *)(comp->fe());
+   bool isLE = comp->target().cpu.isLittleEndian();
+   const bool nonZeroInitial = hashNode != NULL && !hashNode->isConstZeroValue();
+
+
+   TR::Node *valueNode = node->getFirstChild();
+   TR::Node *offsetNode = node->getSecondChild();
+   TR::Node *countNode = node->getThirdChild();
+
+   TR::Register *valueReg = cg->gprClobberEvaluate(valueNode);
+   TR::Register *endReg = cg->gprClobberEvaluate(offsetNode);
+   TR::Register *vendReg = cg->gprClobberEvaluate(countNode);
+   TR::Register *tempReg = cg->allocateRegister();
+   TR::Register *constant0Reg = cg->allocateRegister();
+   TR::Register *multiplierAddrReg = cg->allocateRegister();
+   TR::Register *condReg = cg->allocateRegister(TR_CCR);
+
+   TR::Register *multiplierReg = cg->allocateRegister(TR_VRF);
+   TR::Register *vtmp1Reg = cg->allocateRegister(TR_VRF);
+   TR::Register *vtmp2Reg = cg->allocateRegister(TR_VRF);
+   TR::Register *vconstant0Reg = cg->allocateRegister(TR_VRF);
+   TR::Register *vconstantNegReg = cg->allocateRegister(TR_VRF);
+   TR::Register *vunpackMaskReg = cg->allocateRegister(TR_VRF);
```

Comment:
It seem that this register and some instructions that use it are generated and not used, when type is `Int32` and for all types when `isSigned==true`.